ci/azure/posix.yml

```diff
@@ -24,8 +24,33 @@ jobs:
     - script: echo '##vso[task.prependpath]$(HOME)/miniconda3/bin'
       displayName: 'Set conda path'
 
+    - task: Cache@2
+      displayName: Use cached Anaconda environment
+      inputs:
+        key: 'conda | "$(Agent.OS)" | $(ENV_FILE)'
+        path: $(HOME)/miniconda3/
+        restoreKeys: |
+          python | "$(Agent.OS)"
+          python
+        cacheHitVar: CONDA_CACHE_RESTORED
+
     - script: ci/setup_env.sh
-      displayName: 'Setup environment and build pandas'
+      displayName: 'Setup environment'
+      condition: eq(variables.CONDA_CACHE_RESTORED, 'false')
+
+    - script: |
+        conda update -q -y -n pandas-dev --file $ENV_FILE
+      displayName: 'Update packages in pandas-dev'
+
+    - script: |
+        source activate pandas-dev
+        conda list pandas
+        python setup.py build_ext -q -j2
+        python -m pip install --no-deps -U pip wheel setuptools
+        python -m pip install --no-build-isolation -e .
+        echo "conda list"
+        conda list
+      displayName: 'Build extensions and install pandas'
```

Comment:
Is there a reason you are not using ci/setup_env.sh? Note that that script is also used by other builds on Github Actions, so if you are changing that file, you should gate your changes behind an env variable to avoid it affecting other builds.